R/extractData2.R

```diff
@@ -143,13 +146,34 @@ labels2values2 <- function(dat, labels, convertMiss, dropPartialLabels, labels2c
   change_labels <- labels[labels[, "varName"] %in% convertVariables, ]    # careful, from here use only change_labels!
   # check value labels, remove incomplete labels from insertion to protect variables
   if(identical(dropPartialLabels, TRUE)) {
-    drop_labels <- unlist(lapply(unique(labels$varName), check_labels, dat = dat, labels = labels,
+    drop_labels <- unlist(lapply(unique(labels$varName), FUN = check_labels, dat = dat, labels = labels,
                                  convertMiss = convertMiss))
     change_labels <- change_labels[!change_labels$varName %in% drop_labels, ]
   }
   # early return, if no values are to be recoded
   if(nrow(change_labels) == 0) return(dat)
 
+  # check for duplicate value labels (unfortunately possible in SPSS)
+  vars_with_duplicate_valLabels <- change_labels[duplicated(change_labels[, c("varName", "valLabel")]), "varName"]
+  if(length(vars_with_duplicate_valLabels) > 0) {
+    for(nam in unique(vars_with_duplicate_valLabels)) {
+      single_change_labels <- change_labels[change_labels$varName == nam, ]
+      dup_valLabels <- single_change_labels[duplicated(single_change_labels$valLabel), "valLabel"]
+      affected_values <- single_change_labels[single_change_labels$valLabel == dup_valLabels, ]
+      for(dup_valLabel in dup_valLabels) {
+        warning("Duplicate value label in variable ", nam, ": ", dup_valLabel, ". Information may be lost when extracting data.")
+
```

Comment:
Personally, I like it when warnings are more detailed, even if they sometimes fill up the console. Instead of 
```
Warnmeldungen:
4: In labels2values2(dat = dat, labels = labels, convertMiss = convertMiss,  :
  Duplicate value label in variable VAR1: One. Information may be lost when extracting data.
```
I would prefer
```
Warnmeldung:
Variable 'VAR1' with duplicate value labels (see 'valLabel' column): 
        varName value  valLabel
      1    VAR1   -99 By design
      2    VAR1   -96  Omission
      3    VAR1     1       One
      4    VAR1     2       One
   Only unique 'valLabel' values will be kept if 'VAR1' is extracted with 'labels2factor' option.
``` 
Of course, this is quite redundant, but I think inexperienced users will understand this warning better. If you prefer the short version, you can of course leave it as it is.

Reply:
Adopted your suggestion (and it probably makes sense to adopt this in the future for other `warnings` withing the package). 

Minor modification, however:
The warning only displays the rows with duplicated `valLabels`, as sometimes variables contain a large amount of value labels (e.g., this bug was reported for ISCO variables).